master/buildbot/test/unit/test_worker_openstack.py

```diff
@@ -71,6 +71,17 @@ def test_constructor_minimal(self):
         self.assertEqual(bs.block_devices, None)
         self.assertIsInstance(bs.novaclient, novaclient.Client)
 
+    def test_constructor_minimal_keystone_v3(self):
+        bs = openstack.OpenStackLatentWorker(
+            'bot', 'pass', os_user_domain='test', os_project_domain='test',
+            **self.bs_image_args)
+        self.assertEqual(bs.workername, 'bot')
+        self.assertEqual(bs.password, 'pass')
+        self.assertEqual(bs.flavor, 1)
+        self.assertEqual(bs.image, 'image-uuid')
+        self.assertEqual(bs.block_devices, None)
+        self.assertIsInstance(bs.novaclient, novaclient.Client)
```

Comment:
The auth is saved in Session but that isn't saved in the client. Extend Client to save session so you can assert it.